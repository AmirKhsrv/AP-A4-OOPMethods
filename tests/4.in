show_sorted_shelf 119 read History
recommend_first_approach 5
best_reviewer
credit 118
show_author_info 1276
best_reviewer
recommend_second_approach 79
best_reviewer
credit 83
best_book
recommend_first_approach 81
show_author_info 181
credit 30
best_book
show_author_info 1570
recommend_second_approach 11
show_author_info 1626
show_sorted_shelf 85 currently_reading Fiction
show_author_info 1265
recommend_second_approach 45
best_book
best_reviewer
recommend_second_approach 4
best_book
credit 115
