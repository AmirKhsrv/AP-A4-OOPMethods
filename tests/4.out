id: 193
Title: Eternity Road
Genre: Science Fiction
Author: Jack McDevitt
***
id: 83
Title: Innocents
Genre: New Adult
Author: Mary  Elizabeth
***
id: 335
Title: Painless
Genre: Romance
Author: Devon Hartford
***
id: 167
Title: Shaedes of Gray
Genre: Fantasy
Author: Amanda Bonilla
***
id: 39
Title: Tell Me Three Things
Genre: Young Adult
Author: Julie Buxbaum
***
id: 357
Title: Dragon Rider
Genre: Fantasy
Author: Cornelia Funke
Reviews:
id: 237 Rating: 1 Likes: 0 Date: March 19 2019
id: 807 Rating: 0 Likes: 0 Date: August 25 2016
id: 1205 Rating: 5 Likes: 0 Date: March 14 2018
id: 2621 Rating: 4 Likes: 0 Date: July 14 2017
id: 2748 Rating: 0 Likes: 0 Date: March 9 2016
id: 3126 Rating: 1 Likes: 0 Date: September 6 2017
id: 3394 Rating: 2 Likes: 0 Date: February 21 2016
id: 3594 Rating: 5 Likes: 0 Date: July 26 2018
id: 3689 Rating: 2 Likes: 0 Date: September 19 2016
id: 3908 Rating: 4 Likes: 0 Date: August 29 2019
id: 4156 Rating: 2 Likes: 10 Date: November 19 2017
id: 4336 Rating: 0 Likes: 52 Date: March 13 2019
id: 4661 Rating: 3 Likes: 0 Date: May 4 2015
id: 5057 Rating: 4 Likes: 0 Date: August 28 2019
id: 6502 Rating: 4 Likes: 5 Date: August 1 2016
id: 6702 Rating: 1 Likes: 0 Date: January 22 2017
id: 6756 Rating: 5 Likes: 6 Date: January 24 2018
id: 6811 Rating: 2 Likes: 0 Date: September 1 2019
id: 6824 Rating: 4 Likes: 10 Date: July 10 2016
id: 8132 Rating: 4 Likes: 0 Date: August 14 2015
id: 8520 Rating: 0 Likes: 0 Date: October 2 2017
id: 8965 Rating: 3 Likes: 0 Date: May 17 2018
id: 8986 Rating: 1 Likes: 0 Date: November 18 2016
id: 9012 Rating: 3 Likes: 0 Date: May 16 2019
id: 9278 Rating: 2 Likes: 146 Date: February 29 2013
id: 9309 Rating: 4 Likes: 0 Date: January 10 2017
id: 9758 Rating: 4 Likes: 0 Date: April 13 2018
id: 10411 Rating: 5 Likes: 0 Date: January 13 2019
id: 11475 Rating: 3 Likes: 0 Date: November 16 2019
id: 12798 Rating: 4 Likes: 0 Date: February 24 2019
id: 12866 Rating: 5 Likes: 0 Date: May 26 2014
id: 13003 Rating: 2 Likes: 0 Date: September 20 2015
id: 13316 Rating: 4 Likes: 0 Date: August 15 2016
id: 13330 Rating: 0 Likes: 1 Date: June 18 2015
id: 13589 Rating: 2 Likes: 0 Date: November 11 2015
id: 14238 Rating: 4 Likes: 4 Date: January 16 2012
id: 14588 Rating: 0 Likes: 0 Date: July 2 2016
id: 14619 Rating: 0 Likes: 0 Date: December 9 2019
id: 14678 Rating: 3 Likes: 1 Date: January 18 2019
id: 14985 Rating: 0 Likes: 0 Date: October 14 2019
id: 15249 Rating: 3 Likes: 0 Date: December 29 2019
id: 15753 Rating: 0 Likes: 1 Date: April 22 2018
id: 16044 Rating: 1 Likes: 1 Date: March 17 2017
id: 16148 Rating: 4 Likes: 0 Date: July 16 2019
id: 16478 Rating: 5 Likes: 0 Date: February 26 2014
id: 16649 Rating: 2 Likes: 0 Date: October 8 2016
id: 16992 Rating: 5 Likes: 0 Date: August 1 2016
id: 17039 Rating: 4 Likes: 0 Date: March 6 2019
id: 17101 Rating: 5 Likes: 0 Date: January 14 2016
id: 17406 Rating: 5 Likes: 0 Date: March 21 2014
id: 31
Name: Athena Souri
Place of Birth: Switzerland
Member Since: December 24 2010
Favorite Genres: Romance
Favorite Authors: Carl Sagan, Colleen Clayton, Greer Noble
Number of Books in Read Shelf: 8
Number of Books in Want to Read Shelf: 15
Number of Books in Currently Reading Shelf: 6
Number of Likes: 2455
0.007248
id: 1276
Name: Mira W.
Year of Birth: 1950
Place of Birth: Zambia
Member Since: March 19 2011
Genres: Fiction
Books:
id: 31
Name: Athena Souri
Place of Birth: Switzerland
Member Since: December 24 2010
Favorite Genres: Romance
Favorite Authors: Carl Sagan, Colleen Clayton, Greer Noble
Number of Books in Read Shelf: 8
Number of Books in Want to Read Shelf: 15
Number of Books in Currently Reading Shelf: 6
Number of Likes: 2455
id: 211
Title: The Fall of Five
Genre: Young Adult
Author: Pittacus Lore
Reviews:
id: 76 Rating: 5 Likes: 0 Date: August 19 2016
id: 490 Rating: 4 Likes: 0 Date: January 26 2018
id: 1006 Rating: 1 Likes: 3 Date: January 25 2017
id: 1861 Rating: 4 Likes: 0 Date: February 9 2018
id: 2686 Rating: 0 Likes: 0 Date: September 11 2016
id: 2727 Rating: 5 Likes: 0 Date: November 26 2015
id: 3688 Rating: 2 Likes: 0 Date: September 9 2017
id: 3734 Rating: 1 Likes: 0 Date: April 15 2012
id: 4304 Rating: 3 Likes: 26 Date: September 21 2011
id: 4478 Rating: 4 Likes: 1 Date: March 27 2019
id: 4593 Rating: 5 Likes: 0 Date: January 16 2015
id: 4677 Rating: 4 Likes: 1 Date: September 1 2016
id: 4857 Rating: 5 Likes: 4 Date: September 19 2017
id: 5397 Rating: 5 Likes: 0 Date: October 10 2018
id: 6087 Rating: 1 Likes: 0 Date: June 9 2011
id: 6101 Rating: 4 Likes: 0 Date: March 9 2019
id: 7230 Rating: 1 Likes: 10 Date: December 29 2017
id: 7660 Rating: 5 Likes: 3 Date: April 24 2018
id: 8166 Rating: 5 Likes: 0 Date: January 6 2015
id: 8699 Rating: 1 Likes: 0 Date: July 25 2019
id: 9124 Rating: 0 Likes: 0 Date: February 9 2016
id: 9144 Rating: 3 Likes: 0 Date: December 15 2017
id: 9730 Rating: 1 Likes: 0 Date: June 9 2011
id: 9892 Rating: 2 Likes: 0 Date: May 27 2017
id: 10053 Rating: 5 Likes: 0 Date: October 12 2018
id: 10413 Rating: 5 Likes: 0 Date: December 14 2018
id: 10697 Rating: 0 Likes: 22 Date: April 26 2017
id: 10906 Rating: 0 Likes: 0 Date: October 8 2014
id: 10993 Rating: 1 Likes: 0 Date: October 20 2018
id: 11111 Rating: 2 Likes: 3 Date: February 28 2016
id: 11330 Rating: 0 Likes: 26 Date: September 29 2014
id: 11761 Rating: 5 Likes: 0 Date: April 24 2015
id: 11958 Rating: 3 Likes: 0 Date: October 23 2014
id: 12401 Rating: 1 Likes: 0 Date: August 13 2015
id: 12622 Rating: 0 Likes: 0 Date: November 12 2019
id: 13928 Rating: 2 Likes: 0 Date: November 29 2019
id: 14234 Rating: 2 Likes: 2 Date: March 15 2019
id: 14242 Rating: 1 Likes: 1 Date: April 4 2019
id: 14568 Rating: 0 Likes: 0 Date: February 27 2018
id: 16016 Rating: 3 Likes: 0 Date: December 20 2015
id: 16289 Rating: 4 Likes: 0 Date: August 14 2015
id: 16570 Rating: 4 Likes: 0 Date: October 13 2018
id: 16628 Rating: 1 Likes: 0 Date: January 10 2016
id: 16765 Rating: 4 Likes: 0 Date: October 27 2018
id: 31
Name: Athena Souri
Place of Birth: Switzerland
Member Since: December 24 2010
Favorite Genres: Romance
Favorite Authors: Carl Sagan, Colleen Clayton, Greer Noble
Number of Books in Read Shelf: 8
Number of Books in Want to Read Shelf: 15
Number of Books in Currently Reading Shelf: 6
Number of Likes: 2455
0.005660
id: 405
Title: Ellen Foster
Genre: Fiction
Author: Kaye Gibbons
Average Rating: 3.42
id: 357
Title: Dragon Rider
Genre: Fantasy
Author: Cornelia Funke
Reviews:
id: 237 Rating: 1 Likes: 0 Date: March 19 2019
id: 807 Rating: 0 Likes: 0 Date: August 25 2016
id: 1205 Rating: 5 Likes: 0 Date: March 14 2018
id: 2621 Rating: 4 Likes: 0 Date: July 14 2017
id: 2748 Rating: 0 Likes: 0 Date: March 9 2016
id: 3126 Rating: 1 Likes: 0 Date: September 6 2017
id: 3394 Rating: 2 Likes: 0 Date: February 21 2016
id: 3594 Rating: 5 Likes: 0 Date: July 26 2018
id: 3689 Rating: 2 Likes: 0 Date: September 19 2016
id: 3908 Rating: 4 Likes: 0 Date: August 29 2019
id: 4156 Rating: 2 Likes: 10 Date: November 19 2017
id: 4336 Rating: 0 Likes: 52 Date: March 13 2019
id: 4661 Rating: 3 Likes: 0 Date: May 4 2015
id: 5057 Rating: 4 Likes: 0 Date: August 28 2019
id: 6502 Rating: 4 Likes: 5 Date: August 1 2016
id: 6702 Rating: 1 Likes: 0 Date: January 22 2017
id: 6756 Rating: 5 Likes: 6 Date: January 24 2018
id: 6811 Rating: 2 Likes: 0 Date: September 1 2019
id: 6824 Rating: 4 Likes: 10 Date: July 10 2016
id: 8132 Rating: 4 Likes: 0 Date: August 14 2015
id: 8520 Rating: 0 Likes: 0 Date: October 2 2017
id: 8965 Rating: 3 Likes: 0 Date: May 17 2018
id: 8986 Rating: 1 Likes: 0 Date: November 18 2016
id: 9012 Rating: 3 Likes: 0 Date: May 16 2019
id: 9278 Rating: 2 Likes: 146 Date: February 29 2013
id: 9309 Rating: 4 Likes: 0 Date: January 10 2017
id: 9758 Rating: 4 Likes: 0 Date: April 13 2018
id: 10411 Rating: 5 Likes: 0 Date: January 13 2019
id: 11475 Rating: 3 Likes: 0 Date: November 16 2019
id: 12798 Rating: 4 Likes: 0 Date: February 24 2019
id: 12866 Rating: 5 Likes: 0 Date: May 26 2014
id: 13003 Rating: 2 Likes: 0 Date: September 20 2015
id: 13316 Rating: 4 Likes: 0 Date: August 15 2016
id: 13330 Rating: 0 Likes: 1 Date: June 18 2015
id: 13589 Rating: 2 Likes: 0 Date: November 11 2015
id: 14238 Rating: 4 Likes: 4 Date: January 16 2012
id: 14588 Rating: 0 Likes: 0 Date: July 2 2016
id: 14619 Rating: 0 Likes: 0 Date: December 9 2019
id: 14678 Rating: 3 Likes: 1 Date: January 18 2019
id: 14985 Rating: 0 Likes: 0 Date: October 14 2019
id: 15249 Rating: 3 Likes: 0 Date: December 29 2019
id: 15753 Rating: 0 Likes: 1 Date: April 22 2018
id: 16044 Rating: 1 Likes: 1 Date: March 17 2017
id: 16148 Rating: 4 Likes: 0 Date: July 16 2019
id: 16478 Rating: 5 Likes: 0 Date: February 26 2014
id: 16649 Rating: 2 Likes: 0 Date: October 8 2016
id: 16992 Rating: 5 Likes: 0 Date: August 1 2016
id: 17039 Rating: 4 Likes: 0 Date: March 6 2019
id: 17101 Rating: 5 Likes: 0 Date: January 14 2016
id: 17406 Rating: 5 Likes: 0 Date: March 21 2014
id: 181
Name: Mike Gold
Year of Birth: 1977
Place of Birth: Netherlands
Member Since: April 9 2013
Genres: Comics Graphic Novels
Books:
0.006324
id: 405
Title: Ellen Foster
Genre: Fiction
Author: Kaye Gibbons
Average Rating: 3.42
id: 1570
Name: Cheryl Yeko
Year of Birth: 1993
Place of Birth: Panama
Member Since: July 7 2011
Genres: Romance
Books:
id: 303
Title: The Good Luck Charm
Genre: Romance
Author: Helena Hunting
Reviews:
id: 99 Rating: 3 Likes: 8 Date: November 16 2018
id: 142 Rating: 4 Likes: 0 Date: January 20 2019
id: 949 Rating: 1 Likes: 1 Date: May 14 2017
id: 1209 Rating: 1 Likes: 0 Date: August 22 2013
id: 1350 Rating: 5 Likes: 2 Date: September 25 2018
id: 1595 Rating: 1 Likes: 0 Date: July 1 2013
id: 1747 Rating: 1 Likes: 0 Date: November 13 2018
id: 1846 Rating: 4 Likes: 0 Date: July 9 2016
id: 3401 Rating: 3 Likes: 0 Date: January 1 2016
id: 3889 Rating: 5 Likes: 0 Date: October 17 2017
id: 4887 Rating: 5 Likes: 3 Date: May 4 2018
id: 5127 Rating: 5 Likes: 0 Date: November 9 2010
id: 5574 Rating: 3 Likes: 0 Date: October 6 2014
id: 6137 Rating: 3 Likes: 0 Date: June 28 2017
id: 6480 Rating: 1 Likes: 1 Date: May 1 2011
id: 7559 Rating: 0 Likes: 7 Date: August 19 2017
id: 7913 Rating: 3 Likes: 0 Date: July 10 2019
id: 8048 Rating: 2 Likes: 0 Date: December 23 2018
id: 8160 Rating: 1 Likes: 0 Date: February 26 2017
id: 8310 Rating: 2 Likes: 0 Date: November 29 2019
id: 8626 Rating: 5 Likes: 0 Date: June 16 2014
id: 8683 Rating: 4 Likes: 0 Date: June 20 2013
id: 8858 Rating: 0 Likes: 0 Date: December 16 2015
id: 8862 Rating: 4 Likes: 0 Date: August 26 2018
id: 8903 Rating: 5 Likes: 0 Date: July 11 2018
id: 9182 Rating: 2 Likes: 0 Date: October 12 2018
id: 10427 Rating: 2 Likes: 0 Date: July 28 2018
id: 10432 Rating: 3 Likes: 0 Date: January 9 2018
id: 10490 Rating: 4 Likes: 0 Date: October 20 2017
id: 11047 Rating: 2 Likes: 24 Date: October 15 2010
id: 12102 Rating: 2 Likes: 3 Date: April 22 2018
id: 12147 Rating: 4 Likes: 2 Date: December 12 2019
id: 12633 Rating: 3 Likes: 0 Date: March 21 2016
id: 13302 Rating: 4 Likes: 0 Date: April 18 2015
id: 14111 Rating: 3 Likes: 1 Date: September 14 2016
id: 14698 Rating: 2 Likes: 0 Date: October 22 2019
id: 16634 Rating: 2 Likes: 0 Date: October 13 2018
id: 1626
Name: Mary  Weber
Year of Birth: 1957
Place of Birth: Luxembourg
Member Since: October 9 2015
Genres: Young Adult
Books:
id: 135
Title: Fried Green Tomatoes at the Whistle Stop Cafe
Genre: Fiction
Author: Fannie Flagg
***
id: 357
Title: Dragon Rider
Genre: Fantasy
Author: Cornelia Funke
***
id: 41
Title: If You Find Me
Genre: Young Adult
Author: Emily Murdoch
***
id: 150
Title: The Year of the Great Seventh
Genre: Young Adult
Author: Teresa Orts
***
id: 1265
Name: Chris O'Mara
Year of Birth: 1968
Place of Birth: Tunisia
Member Since: September 10 2010
Genres: Fantasy
Books:
id: 303
Title: The Good Luck Charm
Genre: Romance
Author: Helena Hunting
Reviews:
id: 99 Rating: 3 Likes: 8 Date: November 16 2018
id: 142 Rating: 4 Likes: 0 Date: January 20 2019
id: 949 Rating: 1 Likes: 1 Date: May 14 2017
id: 1209 Rating: 1 Likes: 0 Date: August 22 2013
id: 1350 Rating: 5 Likes: 2 Date: September 25 2018
id: 1595 Rating: 1 Likes: 0 Date: July 1 2013
id: 1747 Rating: 1 Likes: 0 Date: November 13 2018
id: 1846 Rating: 4 Likes: 0 Date: July 9 2016
id: 3401 Rating: 3 Likes: 0 Date: January 1 2016
id: 3889 Rating: 5 Likes: 0 Date: October 17 2017
id: 4887 Rating: 5 Likes: 3 Date: May 4 2018
id: 5127 Rating: 5 Likes: 0 Date: November 9 2010
id: 5574 Rating: 3 Likes: 0 Date: October 6 2014
id: 6137 Rating: 3 Likes: 0 Date: June 28 2017
id: 6480 Rating: 1 Likes: 1 Date: May 1 2011
id: 7559 Rating: 0 Likes: 7 Date: August 19 2017
id: 7913 Rating: 3 Likes: 0 Date: July 10 2019
id: 8048 Rating: 2 Likes: 0 Date: December 23 2018
id: 8160 Rating: 1 Likes: 0 Date: February 26 2017
id: 8310 Rating: 2 Likes: 0 Date: November 29 2019
id: 8626 Rating: 5 Likes: 0 Date: June 16 2014
id: 8683 Rating: 4 Likes: 0 Date: June 20 2013
id: 8858 Rating: 0 Likes: 0 Date: December 16 2015
id: 8862 Rating: 4 Likes: 0 Date: August 26 2018
id: 8903 Rating: 5 Likes: 0 Date: July 11 2018
id: 9182 Rating: 2 Likes: 0 Date: October 12 2018
id: 10427 Rating: 2 Likes: 0 Date: July 28 2018
id: 10432 Rating: 3 Likes: 0 Date: January 9 2018
id: 10490 Rating: 4 Likes: 0 Date: October 20 2017
id: 11047 Rating: 2 Likes: 24 Date: October 15 2010
id: 12102 Rating: 2 Likes: 3 Date: April 22 2018
id: 12147 Rating: 4 Likes: 2 Date: December 12 2019
id: 12633 Rating: 3 Likes: 0 Date: March 21 2016
id: 13302 Rating: 4 Likes: 0 Date: April 18 2015
id: 14111 Rating: 3 Likes: 1 Date: September 14 2016
id: 14698 Rating: 2 Likes: 0 Date: October 22 2019
id: 16634 Rating: 2 Likes: 0 Date: October 13 2018
id: 405
Title: Ellen Foster
Genre: Fiction
Author: Kaye Gibbons
Average Rating: 3.42
id: 31
Name: Athena Souri
Place of Birth: Switzerland
Member Since: December 24 2010
Favorite Genres: Romance
Favorite Authors: Carl Sagan, Colleen Clayton, Greer Noble
Number of Books in Read Shelf: 8
Number of Books in Want to Read Shelf: 15
Number of Books in Currently Reading Shelf: 6
Number of Likes: 2455
id: 211
Title: The Fall of Five
Genre: Young Adult
Author: Pittacus Lore
Reviews:
id: 76 Rating: 5 Likes: 0 Date: August 19 2016
id: 490 Rating: 4 Likes: 0 Date: January 26 2018
id: 1006 Rating: 1 Likes: 3 Date: January 25 2017
id: 1861 Rating: 4 Likes: 0 Date: February 9 2018
id: 2686 Rating: 0 Likes: 0 Date: September 11 2016
id: 2727 Rating: 5 Likes: 0 Date: November 26 2015
id: 3688 Rating: 2 Likes: 0 Date: September 9 2017
id: 3734 Rating: 1 Likes: 0 Date: April 15 2012
id: 4304 Rating: 3 Likes: 26 Date: September 21 2011
id: 4478 Rating: 4 Likes: 1 Date: March 27 2019
id: 4593 Rating: 5 Likes: 0 Date: January 16 2015
id: 4677 Rating: 4 Likes: 1 Date: September 1 2016
id: 4857 Rating: 5 Likes: 4 Date: September 19 2017
id: 5397 Rating: 5 Likes: 0 Date: October 10 2018
id: 6087 Rating: 1 Likes: 0 Date: June 9 2011
id: 6101 Rating: 4 Likes: 0 Date: March 9 2019
id: 7230 Rating: 1 Likes: 10 Date: December 29 2017
id: 7660 Rating: 5 Likes: 3 Date: April 24 2018
id: 8166 Rating: 5 Likes: 0 Date: January 6 2015
id: 8699 Rating: 1 Likes: 0 Date: July 25 2019
id: 9124 Rating: 0 Likes: 0 Date: February 9 2016
id: 9144 Rating: 3 Likes: 0 Date: December 15 2017
id: 9730 Rating: 1 Likes: 0 Date: June 9 2011
id: 9892 Rating: 2 Likes: 0 Date: May 27 2017
id: 10053 Rating: 5 Likes: 0 Date: October 12 2018
id: 10413 Rating: 5 Likes: 0 Date: December 14 2018
id: 10697 Rating: 0 Likes: 22 Date: April 26 2017
id: 10906 Rating: 0 Likes: 0 Date: October 8 2014
id: 10993 Rating: 1 Likes: 0 Date: October 20 2018
id: 11111 Rating: 2 Likes: 3 Date: February 28 2016
id: 11330 Rating: 0 Likes: 26 Date: September 29 2014
id: 11761 Rating: 5 Likes: 0 Date: April 24 2015
id: 11958 Rating: 3 Likes: 0 Date: October 23 2014
id: 12401 Rating: 1 Likes: 0 Date: August 13 2015
id: 12622 Rating: 0 Likes: 0 Date: November 12 2019
id: 13928 Rating: 2 Likes: 0 Date: November 29 2019
id: 14234 Rating: 2 Likes: 2 Date: March 15 2019
id: 14242 Rating: 1 Likes: 1 Date: April 4 2019
id: 14568 Rating: 0 Likes: 0 Date: February 27 2018
id: 16016 Rating: 3 Likes: 0 Date: December 20 2015
id: 16289 Rating: 4 Likes: 0 Date: August 14 2015
id: 16570 Rating: 4 Likes: 0 Date: October 13 2018
id: 16628 Rating: 1 Likes: 0 Date: January 10 2016
id: 16765 Rating: 4 Likes: 0 Date: October 27 2018
id: 405
Title: Ellen Foster
Genre: Fiction
Author: Kaye Gibbons
Average Rating: 3.42
0.005495
